core/src/main/java/com/sequenceiq/cloudbreak/service/credential/CredentialService.java

```diff
@@ -72,135 +87,126 @@
     @Inject
     private OrganizationService organizationService;
 
-    public Set<Credential> retrievePrivateCredentials(IdentityUser user) {
-        return credentialRepository.findForUser(user.getUserId());
+    public Set<Credential> listForUsersDefaultOrganization() {
+        return credentialRepository.findActiveForOrganizationFilterByPlatforms(getDefaultOrg().getId(), accountPreferencesService.enabledPlatforms());
     }
 
-    public Set<Credential> retrieveAccountCredentials(IdentityUser user) {
-        Set<String> platforms = accountPreferencesService.enabledPlatforms();
-        return user.getRoles().contains(IdentityUserRole.ADMIN)
-                ? credentialRepository.findAllInAccountAndFilterByPlatforms(user.getAccount(), platforms)
-                : credentialRepository.findPublicInAccountForUserFilterByPlatforms(user.getUserId(), user.getAccount(), platforms);
+    public Set<Credential> listAvailablesByOrganizationId(Long orgId) {
+        return credentialRepository.findActiveForOrganizationFilterByPlatforms(orgId, accountPreferencesService.enabledPlatforms());
     }
 
     public Credential get(Long id) {
-        return credentialRepository.findById(id)
-                .orElseThrow(accessDenied(String.format("Access is denied: Credential not found by id '%d'.", id)));
+        return Optional.ofNullable(credentialRepository.findActiveByIdAndOrganizationFilterByPlatforms(id, getDefaultOrg().getId(),
+                accountPreferencesService.enabledPlatforms())).orElseThrow(accessDenied(String.format(ACCESS_DENIED_FORMAT_MESS_ID, id)));
     }
 
-    public Supplier<AccessDeniedException> accessDenied(String accessDeniedMessage) {
-        return () -> new AccessDeniedException(accessDeniedMessage);
+    public Credential getByNameForOrganization(String name) {
+        return getByNameForOrganizationId(name, getDefaultOrg().getId());
     }
 
-    public Credential get(Long id, String account) {
-        return Optional.ofNullable(credentialRepository.findByIdInAccount(id, account))
-                .orElseThrow(accessDenied(String.format("Access is denied: Credential not found by id '%d' in %s account.", id, account)));
+    @Override
+    public Credential getByNameForOrganizationId(String name, Long organizationId) {
+        try {
+            return super.getByNameForOrganizationId(name, organizationId);
+        } catch (NotFoundException ignore) {
+            throw accessDenied(String.format(ACCESS_DENIED_FORMAT_MESS_NAME, name)).get();
```

Comment:
Not found exceptions are mapped to 403 forbidden in the NotFoundExceptionMapper so this is unecessary

Reply:
I'm catching the exception just to make a more precise message to the user